neptune/query/query.go

```diff
@@ -112,11 +112,10 @@ const (
 	CreateObservationPart          = `g.addV('_%s_observation').as('o').property(single, 'value', '%s').property(single, 'rowIndex', '%d')`
 	AddObservationRelationshipPart = `.V().hasId('%s').hasLabel('_%s_%s').where(values('value').is("%s")).addE('isValueOf').from('o')`
 
-	GetInstanceHeaderPart  = `g.V().hasLabel('_%s_Instance').values('header').aggregate('results')`
-	GetAllObservationsPart = `.V().hasLabel('_%s_observation').values('row')`
-
-	GetFirstDimensionPart       = `.V().has('_%s_%s','value',within(%s)).in('isValueOf')`
-	GetObservationDimensionPart = `__.as('row').out('isValueOf').has('_%s_%s','value',within(%s))`
-	GetObservationSelectRowPart = `.select('row').by('value').aggregate('results').cap('results').unfold()`
+	GetInstanceHeaderPart       = `g.V().hasLabel('_%s_Instance').values('header').aggregate('results')`
+	GetAllObservationsPart      = `.V().hasLabel('_%s_observation').values('row')`
+	GetFirstDimensionPart       = `.V().hasId(%s).in('isValueOf')`
+	GetAdditionalDimensionsPart = `.where(out('isValueOf').hasId(%s).fold().count(local).is_(%d))`
```

Comment:
%d is a number right? but it looks like we're passing in the addtionalDimensionOptions array? is it meant to be the length? Also, I dont understand the purpose of the count here, is that worth a comment?

Reply:
Correct it is a number. The `addtionalDimensionOptions` array is used for the first parameter `.hasId(%s)`. I then pass in the `additionalDimensions` int value to the second `.count(local).is_(%d)`